src/promptflow/promptflow/core/_flow.py

```diff
@@ -52,12 +65,15 @@ def load(
         """
         flow_dir, flow_filename = resolve_flow_path(source)
         flow_path = flow_dir / flow_filename
-        with open(flow_path, "r", encoding=DEFAULT_ENCODING) as f:
-            flow_content = f.read()
-            data = load_yaml_string(flow_content)
-        if is_flex_flow(yaml_dict=data, working_dir=flow_dir):
-            raise UserErrorException("Please call entry directly for flex flow.")
-        return cls._create(code=flow_dir, path=flow_path, data=data)
+        if is_prompty_flow(flow_path):
+            return Prompty.load(source=flow_path, **kwargs)
+        else:
+            with open(flow_path, "r", encoding=DEFAULT_ENCODING) as f:
```

Comment:
Do we have a function called load_yaml_file?

Reply:
There is no such utility function. This logic is also not suitable for extraction, and some calls need to be processed with content